File: Back/Microservices/StockService/src/main/java/com/PFE/stock/services/interfaces/ComboService.java
package com.PFE.stock.services.interfaces;

import com.PFE.stock.entities.Combo;
import com.PFE.stock.services.Exceptions.ComboNotFoundException;
import com.PFE.stock.services.Exceptions.DuplicateComboException;

import java.util.List;

public interface ComboService {
    Combo createCombo(Combo combo) throws DuplicateComboException;
    Combo updateCombo(Integer id, Combo combo) throws ComboNotFoundException, DuplicateComboException;
    Combo getComboById(Integer id) throws ComboNotFoundException;
    List<Combo> getAllCombosByEstablishmentId(Integer establishmentId);
    public void deleteCombo(Integer comboId) throws ComboNotFoundException;;
}

File: Front/src/app/demo/extra/combo/comboService.ts
import { Injectable } from "@angular/core";
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { throwError } from 'rxjs';
import { Combo } from "../Stock/service/models";



@Injectable({
    providedIn: 'root',
})
export class ComboService {
    private comboUrl = 'http://localhost:8083/STOCK/combo';



    constructor(private http: HttpClient) { }


    token = localStorage.getItem("token")



    httpOptions = {
        headers: new HttpHeaders({
            'Authorization': `Bearer ${this.token}`
        })
    };


    createCombo(combo: Combo): Observable<Combo> {
        return this.http.post<Combo>(`${this.comboUrl}/add`, combo, this.httpOptions);
    }

    updateCombo(id: number, combo: Combo): Observable<Combo> {
        return this.http.put<Combo>(`${this.comboUrl}/update/${id}`, combo, this.httpOptions);
    }

    getComboById(id: number): Observable<Combo> {
        return this.http.get<Combo>(`${this.comboUrl}/getBy/${id}`, this.httpOptions);
    }

    getAllCombosByEstablishmentId(establishmentId: number): Observable<Combo[]> {
        return this.http.get<Combo[]>(`${this.comboUrl}/GetALL/${establishmentId}`, this.httpOptions);
    }

    deleteCombo(id: number): Observable<any> {
        return this.http.delete<any>(`${this.comboUrl}/combos/${id}`, this.httpOptions);
    }




}